tools/testdxildll/DxilValidator.cpp

```diff
@@ -0,0 +1,236 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilValidator.cpp                                                         //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements the DirectX Validator object for testing dxil lib              //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DxilContainer/DxilContainer.h"
+#include "dxc/HLSL/DxilValidation.h"
+#include "dxc/Support/WinIncludes.h"
+
+#include "dxc/DxilRootSignature/DxilRootSignature.h"
+#include "dxc/Support/FileIOHelper.h"
+#include "dxc/Support/Global.h"
+#include "dxc/Support/dxcapi.impl.h"
+#include "dxc/Support/microcom.h"
+
+#ifdef _WIN32
+#include "dxcetw.h"
+#endif
+
+using namespace llvm;
+using namespace hlsl;
+
+class DxcValidator : public IDxcValidator2, public IDxcVersionInfo {
```

Comment:
Wait, why can't we move DxcValidator.cpp out of dxcompiler project into shared lib, then you can simply use the same one instead reimplementing so much here?

Reply:
Done